src/Eshopworld.Messaging/Core/IMessenger.cs

```diff
@@ -0,0 +1,56 @@
+﻿using System;
+using System.Threading.Tasks;
+using Eshopworld.Core;
+
+namespace Eshopworld.Messaging.Core
+{
+    /// <summary>
+    /// Contract that provides a simple way to send events to topics
+    /// <remarks>
+    /// Custom topic name needs to be provided.
+    /// </remarks>
+    /// </summary>
+    public interface IPublishEventsWithTopicName: IDisposable
```

Comment:
I think you might be going to granular with your interface names; instead consider overloading the methods defined in the existing messaging interfaces.

Reply:
thanks! 
I did this because the internal behaviour is very different. E.g. this creates a topic. The other one creates a topic based on the message type. From my point of view you would be using one behaviour or another. 

Happy to update the existing interface if that makes more sense anyway. What's your view @mayankgoyel @chomick ?

Reply:
I would keep the method name the same for both and just override with the additional signature.

The underlyng action has the same purpose, which is to create a subscription to a topic , the means of doing this is something for the implmentation to abstract for the invoker.

Reply:
Agree with Chris on that

Reply:
this is now updated here => https://github.com/eShopWorld/core/pull/31

Reply:
@C-M-Ryan @chomick It doesn't seem correct to me. We are unnecessary updating an existing interface which will result in a breaking change for "Eshopworld.Core" package. Why would we want to force the consumer of the existing interface to implement both set of the methods when these sets are mutual exclusive. The consumer will either use a version with specific topic name or the version extracting topic name from message type.  
So I don't understand the logic of combining both sets into a single interface. I would keep them as separated interfaces.
That why; it will not be a braking change for existing Core package as well.